pkg/distributor/query.go

```diff
@@ -257,7 +259,9 @@ func (d *Distributor) queryIngesterStream(ctx context.Context, replicationSet ri
 		defer stream.CloseSend() //nolint:errcheck
 
 		for {
-			resp, err := stream.Recv()
+			var resp ingester_client.WrappedQueryStreamResponse
```

Comment:
I would rename the variables to make it clear it's unsafe to retain such memory.

```suggestion
			var unsafeResp ingester_client.WrappedQueryStreamResponse
```

Reply:
I would do the same for the response variable propagated through the chan in this function.